third_party/move/move-model/src/builder/module_builder.rs

```diff
@@ -608,6 +614,32 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
         });
     }
 
+    fn decl_ana_friend_decl(&mut self, friend_mod_id: &EA::ModuleIdent, friend: &EA::Friend) {
+        let address = match friend_mod_id.value.address {
+            EA::Address::Numerical(_, sp!(_, num_address)) => {
+                Address::Numerical(num_address.into_inner())
+            },
+            EA::Address::NamedUnassigned(name) => {
+                let addr_alias = self.symbol_pool().make(name.value.as_str());
+                let symbolic_addr = Address::Symbolic(addr_alias);
+                if let Some(num_address) = self.parent.env.resolve_address_alias(addr_alias) {
+                    Address::Numerical(num_address)
+                } else {
+                    symbolic_addr
+                }
+            },
+        };
+        let friend_mod_sym = self
+            .symbol_pool()
+            .make(friend_mod_id.value.module.0.value.as_str());
+        let friend_mod_name = ModuleName::new(address, friend_mod_sym);
```

Comment:
I get it that this name cannot be resolved here as the module is possibly not yet processed (modules are build in bottom-up dependency order). But we really would like to have `FriendDecl.module_id` instead of `FriendDecl.module_name` because that is the general way how modules are represented in the model.

Achieving this is a bit more work. You would have first build some temporary data structure to keep friend decls (like `BTreeMap<ModuleId, FriendEntry>` in the `ModelBuilder` and than after all modules have been build, patch this against the modules in the env. This can be done e.g. in `run_move_checker` which is the main entry point for v2 compiler model building.

Reply:
To confirm, we don't need to do this for `run_spec_checker` as well because it is deprecated?

Reply:
run_spec_checker is not deprecated but the entry point for the Move prover, but yes, we don't need to do it there, and only for v2, since run_spec_checker gets this info from bytecode.